frontend/app/src/App.tsx

```diff
@@ -1193,11 +1179,20 @@ export class App extends PureComponent<Props, State> {
         scriptRunId,
         scriptName,
         appHash,
-        elements: AppRoot.empty(false, sidebarElements),
+        elements: this.appNavigation.clearPageElements(
+          this.pendingElementsBuffer,
+          mainScriptHash,
+          sidebarElements
+        ),
       },
       () => {
         this.pendingElementsBuffer = this.state.elements
-        this.widgetMgr.removeInactive(new Set([]))
+        const activeWidgetIds = new Set(
+          Array.from(this.state.elements.getElements())
+            .map(element => getElementWidgetID(element))
+            .filter(notUndefined)
```

Comment:
nit: could you add a comment here please about in which scenarios elements can be undefined despite being in the elements list? Or is it just a precautious check?

Reply:
Yes. Will do. For context, this is all elements, and not all elements are widgets. So this is trying to get all the widget ids. For non-widgets, it will just return null/undefined.

Reply:
ahh ok, thanks! yeah I think adding that line as a comment here would be great